include/boost/geometry/algorithms/detail/relate/topology_check.hpp

```diff
@@ -173,6 +173,8 @@ struct topology_check<MultiLinestring, multi_linestring_tag>
 
             std::size_t count = boost::size(ls);
 
+            m_has_interior = false;
```

Comment:
This shouldn't be set here, It should be set outside the loop because otherwise a MultiLinestring won't have an interior if an empty Linestring is the last element of MultiLinestring.